api/src/test/java/com/epam/pipeline/controller/cluster/ClusterControllerTest.java

```diff
@@ -0,0 +1,309 @@
+        Mockito.doReturn(nodeInstances).when(mockClusterApiService).filterNodes(Mockito.refEq(filterNodesVO));
+
+        final MvcResult mvcResult = performRequest(post(FILTER_NODES_URL).content(content));
+
+        Mockito.verify(mockClusterApiService).filterNodes(Mockito.refEq(filterNodesVO));
+        assertResponse(mvcResult, nodeInstances, NodeCreatorUtils.NODE_INSTANCE_LIST_TYPE);
+    }
+
+    @Test
+    public void shouldFailLoadNodeForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(get(String.format(LOAD_NODE_URL, NAME)));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadNode() throws Exception {
+        Mockito.doReturn(nodeInstance).when(mockClusterApiService).getNode(NAME);
+
+        final MvcResult mvcResult = performRequest(get(String.format(LOAD_NODE_URL, NAME)));
+
+        Mockito.verify(mockClusterApiService).getNode(NAME);
+        assertResponse(mvcResult, nodeInstance, NodeCreatorUtils.NODE_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailLoadNodeFilteredForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(post(LOAD_NODES_URL, NAME));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadNodeFiltered() throws Exception {
+        final FilterPodsRequest filterPodsRequest = NodeCreatorUtils.getDefaultFilterPodsRequest();
+        final String content = getObjectMapper().writeValueAsString(filterPodsRequest);
+        Mockito.doReturn(nodeInstance).when(mockClusterApiService)
+                .getNode(Mockito.eq(NAME), Mockito.refEq(filterPodsRequest));
+
+        final MvcResult mvcResult = performRequest(post(String.format(LOAD_NODE_URL, NAME)).content(content));
+
+        Mockito.verify(mockClusterApiService)
+                .getNode(Mockito.eq(NAME), Mockito.refEq(filterPodsRequest));
+        assertResponse(mvcResult, nodeInstance, NodeCreatorUtils.NODE_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailTerminateNodeForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(delete(NODE_NAME_URL, NAME));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldTerminateNode() throws Exception {
+        Mockito.doReturn(nodeInstance).when(mockClusterApiService).terminateNode(NAME);
+
+        final MvcResult mvcResult = performRequest(delete(String.format(NODE_NAME_URL, NAME)));
+
+        Mockito.verify(mockClusterApiService).terminateNode(NAME);
+        assertResponse(mvcResult, nodeInstance, NodeCreatorUtils.NODE_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailLoadAllInstanceTypesForUnauthorizedUser() throws Exception {
+        performUnauthorizedRequest(get(LOAD_INSTANCE_TYPES_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadAllowedInstanceTypes() throws Exception {
+        final MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add("regionId", "1");
```

Comment:
PMD expects us to get rid of all string literals duplications. So could you please extract variables for regionId, toolInstances and etc.